src/plugins/intel_cpu/tests/functional/single_layer_tests/classes/convolution_backprop_data.cpp

```diff
@@ -0,0 +1,252 @@
+}
+
+std::shared_ptr<ov::Model> DeconvolutionLayerCPUTest::createGraph(const std::vector<ov::PartialShape>& inShapes, ngraph::helpers::InputLayerType outShapeType) {
+    ov::ParameterVector params{std::make_shared<ov::op::v0::Parameter>(prec, inShapes.front())};
+    std::shared_ptr<ov::Node> outShapeNode;
+    if (!outShapeData.empty()) {
+        if (outShapeType == ngraph::helpers::InputLayerType::PARAMETER) {
+            IE_ASSERT(inputDynamicShapes.size() == 2);
+            auto outShapeParam = std::make_shared<ngraph::opset8::Parameter>(ngraph::element::i32, inputDynamicShapes.back());
+            params.push_back(outShapeParam);
+            outShapeNode = outShapeParam;
+        } else {
+            outShapeNode = ngraph::opset8::Constant::create(ngraph::element::i32, {outShapeData[inferRequestNum].size()}, outShapeData[inferRequestNum]);
+        }
+    }
+
+    for (size_t i = 0; i < params.size(); i++) {
+        params[i]->set_friendly_name(std::string("param_") + std::to_string(i));
+    }
+
+    std::shared_ptr<ov::Node> deconv;
+    if (!outShapeData.empty()) {
+        IE_ASSERT(outShapeNode != nullptr);
+        deconv = ngraph::builder::makeConvolutionBackpropData(params[0], outShapeNode, prec, kernel, stride, padBegin,
+                                                              padEnd, dilation, padType, convOutChannels);
+    } else {
+        deconv = ngraph::builder::makeConvolutionBackpropData(params[0], prec, kernel, stride, padBegin,
+                                                              padEnd, dilation, padType, convOutChannels, false, outPadding);
+    }
+
+    return makeNgraphFunction(prec, params, deconv, "DeconvCPU");
+}
+
+void DeconvolutionLayerCPUTest::SetUp() {
+    rel_threshold = 1e-4f;
+
+    targetDevice = ov::test::utils::DEVICE_CPU;
+
+    DeconvSpecParams basicParamsSet;
+    DeconvInputData inputData;
+    fusingSpecificParams fusingParams;
+    CPUSpecificParams cpuParams;
+    std::map<std::string, std::string> additionalConfig;
+    std::tie(basicParamsSet, inputData, prec, fusingParams, cpuParams, additionalConfig) = this->GetParam();
+
+    InputShape inputShape;
+    ngraph::helpers::InputLayerType outShapeType;
+    std::tie(inputShape, outShapeType, outShapeData) = inputData;
+
+    configuration.insert(additionalConfig.begin(), additionalConfig.end());
+    std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+    std::tie(postOpMgrPtr, fusedOps) = fusingParams;
+
+    std::tie(kernel, stride, padBegin, padEnd, dilation, convOutChannels, padType, outPadding) = basicParamsSet;
+
+    if (additionalConfig[InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16] ==
+        InferenceEngine::PluginConfigParams::YES) {
+        inType = outType = prec = ElementType::bf16;
+        rel_threshold = 1e-2f;
+    } else {
+        inType = outType = prec;
+    }
+
+    selectedType = makeSelectedTypeStr(selectedType, prec);
+
+    if  (inputShape.second.size() > 1 &&
```

Comment:
Ideally we should create a separate instances for the input shapes which are supported by x64 and not supported by acl, thus we avoid this check.
But this approach have downsides:
1) It is not clear (at least for me) why acl does not support this particular configuration. This is not like acl does not support some specific algorithm, rank, or something. It is hard to justify the separation for such limitations.
2) If we separate the instances and will test only the "proper" shapes for acl, than we never notice, if the limitation is gone and will continue test the limited configuration of the shapes.

@dmitry-gorokhov Could you please provide you opinion on this one?

Reply:
I agree with the concern about separate shape instances per backend
Just an idea: can we pass std::function as test parameter which will initilize expected ptimitive type based on input shapes (and possible other arguments)? So we still have separate instances per backend, but all parameters are defined in common part, except srd::function implementation which is backend specific.

Reply:
We could do it this way, but this contradicts a bit with the initial idea to test a particular implementation regardless of it's priority.
So from my understanding in general we still want to force a particular implementation, but there can be exceptions, when a back-end does not support some specific (tricky?) configuration, which we can handle by some "fallback" logic, like some function, as you mentioned.
Ideally we could localize this logic in some static function in the source code (since it is also required there as part of "isSupported()" anyway), and use it in scope of the tests as well.

Reply:
@maxnick Could please provide your pov

Reply:
We must test the code that is designed to work. Thus, the question is whether these parameter sets are expected to work? If yes, but in the reality they don't, then this is a bug and we create a ticket and disable the test. If it's not expected that some parameters sets are supported by ACL, then we don't have to declare such test case instances and test other implementations.
Another question is how to conveniently generate test cases considering some backend specifics? Actually we can develop generator functions per backend, which generates test case parameter sets considering all the necessary backend specifics (if any).

Reply:
@EgorDuplensky  @dmitry-gorokhov @maxnick corrected, added tests inskip list